nncf/quantization/quantization_metric.py

```diff
@@ -0,0 +1,273 @@
+import numpy as np
+import networkx as nx
+from copy import deepcopy
+from texttable import Texttable
+from collections import deque
+
+from nncf.quantization.layers import SymmetricQuantizer
+from nncf.utils import get_all_modules_by_type
+from nncf.nncf_network import NNCFNetwork
+from nncf.dynamic_graph.transform_graph import is_nncf_module
+
+class BaseMetric:
+    def __init__(self):
+        pass
+
+    def collect(self):
+        pass
+
+    def get_metric_table(self):
+        pass
+
+
+class NetworkQuantizationShareMetric(BaseMetric):
+    """
+    This class collects the quantization metrics:
+    - Percentage of symmetric/asymmetric/per_channel weight quantizers relative
+      to the number of placed weight quantizers
+    - Percentage of symmetric/asymmetric/per_channel weight quantizers relative
+      to the number potential weight quantizers (max possible number)
```

Comment:
The "max possible number" must be clearly defined - under what conditions, in what scenario, with which config options and with what strategy is that max possible number achieved.